api/server/router/container/container_routes.go

```diff
@@ -512,9 +513,29 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		}
 	}
 
+	if hostConfig != nil && versions.LessThan(version, "1.42") {
+		// Ignore BindOptions.CreateMountpoint because it was added in API 1.42.
+		for _, m := range hostConfig.Mounts {
+			if bo := m.BindOptions; bo != nil {
+				bo.CreateMountpoint = false
+			}
+		}
+	}
```

Comment:
Per my comment above;

```suggestion
	if hostConfig != nil && versions.LessThan(version, "1.42") {
		for _, m := range hostConfig.Mounts {
			// These combinations are invalid, but weren't validated in API < 1.42.
			// We reset them here, so that validation doesn't produce an error.
			if o := m.VolumeOptions; o != nil && m.Type != mount.TypeVolume {
				m.VolumeOptions = nil
			}
			if o := m.TmpfsOptions; o != nil && m.Type != mount.TypeTmpfs {
				m.TmpfsOptions = nil
			}
			if bo := m.BindOptions; bo != nil {
				// Ignore BindOptions.CreateMountpoint because it was added in API 1.42.
				bo.CreateMountpoint = false
			}
		}
	}
```

Note that I'm not resetting `m.BindOptions`, as that was actually validated in API < 1.42, so using that with the wrong type should produce an error.